File: python/read_write.py

numbers = []

# open "numbers.txt" in read mode
with open("./data/numbers.txt", "r") as file:
    for line in file:
        numbers.append(int(line))

# open 'results.txt' in write mode
with open("./data/results.txt", "w") as file:
    # decrease every element by 100
    # and save in results.txt
    for elem in numbers:
        elem -= 100
        str_val = str(elem) + "\n"
        file.write(str_val)
        

File: Cpp/ReadWrite/ReadWrite.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main()
{
    fstream file("./numbers.txt", ios::in);
    fstream fileOut("./results.txt", ios::out);
    string line;

    int sum = 0;

    if (!file.is_open()) {
        cout << "error: cannot open file" << endl;
        return 1;
    }
    

    // Read line by line
    while (getline(file, line)){
        int newVal = stoi(line) - 100;
        fileOut << newVal << endl;
    }

    file.close();
    fileOut.close();
}
